aten/src/ATen/native/quantized/cpu/qclamp.cpp

```diff
@@ -14,12 +17,74 @@ namespace native {
 DEFINE_DISPATCH(qclamp_stub);
 
 namespace {
+
+#ifdef USE_PYTORCH_QNNPACK
+Tensor qnnpack_clamp(Tensor input, Scalar min, Scalar max) {
+
+  TORCH_CHECK(input.ndimension() > 0, "qnnpack_clamp(): Got empty input tensor");
+
+  initQNNPACK();
+
+  Tensor input_contig = input.contiguous();
```

Comment:
We recently saw some perf implications of contiguous calls since it does contiguous in NCHW format. Can you use `input.suggest_memory_format()` here?